Reviewed `057d38d8` against `feature/registry-composed-source-walks`. Step 2a of #596.

## Reproduced

Workspace tests (829), strict Clippy, `cargo fmt --check`, rustdoc, the
regeneration check, and `covertest-kotlin`'s 61 sections on the JVM.

The generated change is exactly what the description says: 75 added, 52
removed, in four hunks, each one a nullable property's decode —
`Annotated.priority`, `DurationBoundary.delay`, `Unsigned.maybeLong` and
`Marker.Ranked.v0`. I read all four. The per-property local becomes
`__present`, the property is wrapped in a block that scopes `v`, and the two
kinds keep their difference: the three wrapping properties still emit
`Option::Some(..)`, and `delay` still calls the whole `Option<Duration>`
converter bare.

Splitting this out from the walk is the right call, and the reason given for it
is the real one: a `Product` part's child has to be the whole conversion of
that part, and these two kinds were splitting theirs across the traversal.

## The JVM evidence is stronger than "61 sections pass"

Sections passing would not by itself mean the changed branches ran. Both
branches of all four decodes are exercised:

| decode | absent | present |
|---|---|---|
| `Annotated.priority` | `annotatedNew(…, null, null)` | `annotatedNew(…, Priority.HIGH)` |
| `DurationBoundary.delay` | `DurationBoundary(0uL, null)` | `DurationBoundary(7uL, 12_345uL)` |
| `Unsigned.maybeLong` | `max.copy(maybeLong = null)` | `Unsigned(…, 0uL, …)` |
| `Marker.Ranked.v0` | `Marker.Ranked(null)` | `Marker.Ranked(Priority.HIGH)` |

The null arm and the unbox arm both run for every shape this PR restructures.

## A correction to my own note on #596

I told #596 that the covertest declares `jobject_input()` three times, framing
the whole-object coverage around that. This PR shows the framing was too
narrow: **none of the four classes whose decode changed is declared
`jobject_input()`** — `Annotated` and `Unsigned` are plain `data_class!`, and
`Marker` is a `sealed_class!`.

`planned_struct_codec` builds the whole-object input plan for any named struct
whose crossing carries a `JObject` wire, so the path steps 2 to 4 move is
reached far more widely than by the three explicit declarations. That is good
news for their coverage and worth knowing before step 2b, since it is also the
blast radius.

## The one new decision has no test

`OptionalBridge::source_present` is a defaulted method, and this PR adds the
first override of it. Getting it wrong is not silent — dropping the override
fails the example build with

```
error[E0308]: mismatched types
  expected `Option<Duration>`, found `Option<Option<Duration>>`
```

— which is a better failure mode than #592's `provides` had. But **no unit test
catches it**: with the override neutered, all `prebindgen-jni` tests pass, and
only rustc on the generated crate objects.

`flattened_field_composes_bounded_conversion_stages` looks like the test that
would, and it did get sharper here — naming the niche converter instead of any
`__jni_in_convert_`. It still does not pin the pass-through: it asserts the
converter is called on `__present`, which is equally true when the result is
wrapped. One line beside it closes that, and I checked it both ways — it passes
as the code stands and fails with the override dropped:

```rust
assert!(
    !rc.contains("Some(__jni_in_convert_wire_to_Option_Duration"),
    "the niche converter already yields the Option, so the layer must not wrap it:\n{rust}"
);
```

Worth having because `source_present` will get its second and third overrides
in steps 3 and 4, and the default is the one that silently applies to whoever
does not think about it.

Nothing here blocks the PR.

— Claude Code (Opus 5)

--- a/prebindgen-jni/src/jni/tests/values.rs
+++ b/prebindgen-jni/src/jni/tests/values.rs
@@ -794,6 +794,11 @@ fn flattened_field_composes_bounded_conversion_stages() {
          value the composed Optional bound:\n{rust}"
     );
     assert!(
+        !rc.contains("Some(__jni_in_convert_wire_to_Option_Duration"),
+        "the niche converter already yields the Option, so the composed layer must pass its \
+         answer through rather than wrap it:\n{rust}"
+    );
+    assert!(
         rc.contains("let___delay:jni::sys::jlong=__jni_out_convert_")
             && rc.contains("\"(J)Lio/test/jni/Timed;\""),
         "whole-struct output must pass the niche as primitive jlong:\n{rust}"

